frontend/src/component/MenuList/NavItem/index.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { forwardRef, ForwardRefExoticComponent, RefAttributes, useEffect, useMemo } from 'react';
+import { Link } from 'react-router-dom';
+// assets
+import FiberManualRecordIcon from '@mui/icons-material/FiberManualRecord';
+import { ListItemButton, ListItemIcon, ListItemText, Typography } from '@mui/material';
+
+import { dispatch, useAppSelector } from '../../../store/hooks';
+import { viewsMiddleware, viewsSelector } from '../../../store/slices/menu';
+// types
+import { LinkTarget, NavItemType } from '../../../types';
+
+interface NavItemProps {
+  item: NavItemType;
+  level: number;
+}
+
+// ==============================|| SIDEBAR MENU LIST ITEMS ||============================== //
+
+const NavItem = ({ item, level }: NavItemProps) => {
+  const { openItem } = useAppSelector(viewsSelector.menu);
+
+  const Icon = item?.icon!;
+  const itemIcon = item?.icon ? (
+    <Icon stroke="1.5" color="primary" />
+  ) : (
+    <FiberManualRecordIcon
+      sx={{
+        width: openItem.findIndex((id: number | string) => id === item?.id) > -1 ? 8 : 6,
+        height: openItem.findIndex((id: number | string) => id === item?.id) > -1 ? 8 : 6
+      }}
+      fontSize={level > 0 ? 'inherit' : 'medium'}
+    />
+  );
```

Comment:
**question:** I'm not sure how you do it with components, but I'ld suggest to not use ternar operator when it's cases are big, better to have if/else structure. @RobMin confirm please.